src/components/TwoOptionsToggle.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useState } from 'react';
+import { StyleSheet, View, Text, TouchableOpacity } from 'react-native';
+import { COLORS } from '../styles/themes';
+
+/**
+ * @param {{
+ *   options: Map<'first'|'second', { value: string; onTap: Function; }>;
+ *   defaultOption: 'first'|'second';
+ * }}
+ */
+export const TwoOptionsToggle = ({ options, defaultOption }) => {
+  const [currOption, setCurrOption] = useState(defaultOption);
+  const isActive = (option) => (componentOption) => option === componentOption;
+  const onTap = (option) => {
+    setCurrOption(option);
+    // Execute the callback assigned to the option
+    options[option].onTap();
```

Comment:
I would just add here (or in the beginning of the method) `if (option !== currOption)`.

I know it doesn't make difference for the current `onTap` methods but if we change in the future it might.

Reply:
I didn't understand. How we could change it in the future to change the behavior of this component? The option is param is determined by `Option` component itself and I believe React do the job to check if it get a different result to do another render. Are you concerned of unnecessary rendering?

Reply:
The react checks the rendering but the `onTap` is called directly. Can you double check if this `onTap` method is being called even when the option is already selected?

Reply:
Well, it is called every time, indeed. But I don't see it as a problem because of this:
>If the new value you provide is identical to the current state, as determined by an [Object.is](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is) comparison, React will skip re-rendering the component and its children. This is an optimization. Although in some cases React may still need to call your component before skipping the children, it shouldn’t affect your code.
- see at https://react.dev/reference/react/useState#setstate-caveats

Nevertheless I can implement the check.

Reply:
Done.